### PR TITLE
BUG: Fix zero-sized slicing and improve test coverage

### PR DESCRIPTION
**Pull request by tbirdso**
Addresses issue where time and channel slice sizes were set to zero instead of one. The result of this issue was such that reading from stores with time or channel axes returned correct metadata but all-zero-valued image buffers.

This issue slipped by because I was testing for metadata and voxel array shape correctness, but not the actual content of the images that were produced. I have overhauled testing to cover this case and prevent similar regressions in the future:
- Added baseline images for all HTTP tests and refactored to rely on ITKTestDriver's image comparison for metadata and buffer content correctness at the end of each test.
- Added test to verify that we can read from a small local `t,y,x` zarr store with two time frames and validate that each frame has correct metadata and image content.

Also adds optional debugging output to `OMEZarrNGFFImageIO` to help decipher ITK/tensorstore mismatch issues in the future.

### CODE REVIEW

**Review by thewtex — APPROVED** [review]

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -105,6 +106,32 @@ COMMAND IOOMEZarrNGFFTestDriver
...
 )
 
+itk_add_test(
+  NAME IOOMEZarrNGFF_readTimeIndex0
+  COMMAND IOOMEZarrNGFFTestDriver
+    --compare
+      DATA{Baseline/simple_tyx_t0.mha}
+      ${ITK_TEST_OUTPUT_DIR}/simple_tyx_t0.mha
+    itkOMEZarrNGFFReadSliceTest
+      DATA{Input/simple_tyx.zarr.zip}
```
Comment (dzenanz) [inline test/CMakeLists.txt:116]: You forgot to add this image (`simple_tyx.zarr.zip`) to git.

**Review by dzenanz — COMMENTED** [review]
Mostly looks good.

File: test/CMakeLists.txt (hunk 1 above)
Comment (tbirdso) [reply test/CMakeLists.txt:116]: I see it here in the PR files, is it visible for you? https://github.com/InsightSoftwareConsortium/ITKIOOMEZarrNGFF/pull/49/files#diff-a884acf9d27f8c07da55820013d4cea8ed4b0be4118141e521c1206190d7b4db

Comment (dzenanz) [reply test/CMakeLists.txt:116]: Oops, you are right - I didn't see it there in the middle.

**Review by dzenanz — APPROVED** [review]

**Comment by tbirdso** [comment]
I am able to reproduce test failures on my Linux machine. Will investigate.

**Comment by tbirdso** [comment]
Tests are passing after addressing a small test build warning. (had incorrectly marked a test function with `bool` return type when it should have been `void`.)

Moving forward with merge.

**Pull request merged**